### PR TITLE
Fixed $lookup with displayLanguage caching issue

### PR DESCRIPTION
**Pull request by frankjtao**
This is an addendum to https://github.com/hapifhir/hapi-fhir/pull/2908

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4CodeSystemDesignationTest.java [hunk 1]
```diff
@@ -159,6 +159,171 @@ public void testLookupWithoutDisplayLanguage() {
...
+		assertEquals("fr-FR:Systolic blood pressure 12 hour minimum", designation.getPart().get(2).getValue().toString());
+
+		// 3. Systolic blood pressure 12 hour minimum (no language)
+		designation = designationList.get(1);
+		assertEquals("language", designation.getPart().get(0).getName());
+		assertNull(designation.getPart().get(0).getValue());
+		assertEquals("value", designation.getPart().get(2).getName());
+		assertEquals("Systolic blood pressure 12 hour minimum", designation.getPart().get(2).getValue().toString());
+	}
+
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4CodeSystemDesignationTest.java:326]: This is a 150 line test which is pretty unreadable. I'm still not sure exactly what it's doing. Can you possibly extract some of the setup? Do you need 4 separate calls to the service to replicate this bug?

**Review by tadgh — APPROVED** [review]
Approved pending two things: 
1. Needs a changelog. 
2. Please modify this MR and explain in plain english the scenario in which this bug would manifest. I assume its related to cache failures due to language, but there is zero description on this MR, and it doesn't look like there is an associated issue. 
3. Please try to shrink that test if at all possible, as it is >150 lines which makes it tough to maintain.

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4CodeSystemDesignationTest.java (hunk 1 above)
Comment (frankjtao) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/provider/r4/ResourceProviderR4CodeSystemDesignationTest.java:326]: @tadgh 
1/2: This bug is related to the PR https://github.com/hapifhir/hapi-fhir/pull/2908 to support displayLanguage 
For the $lookup operation, the result was cached based system+code. To support displayLanguage, the the caching should be based on system+code+displayLanguage. I think this is part of displayLanguage implementation. Still need a changelog?
3. I will optimize the test case.
